server/webapp/WEB-INF/rails.new/app/controllers/api_v1/admin/environments_controller.rb

```diff
@@ -54,7 +54,18 @@ def patch
         agents_to_add = agents[:add] || []
         agents_to_remove = agents[:remove] || []
 
-        environment_config_service.patchEnvironment(@environment_config, pipelines_to_add, pipelines_to_remove, agents_to_add, agents_to_remove, current_user, result)
+        env_vars = params[:environment_variables] || {}
```

Comment:
We're changing the contract of an API, this is not something that should be done. EVER. Introduce a new API version if you must.